backend/forecastbox/config.py

```diff
@@ -157,12 +166,53 @@ class FIABConfig(BaseSettings):
     model_config = SettingsConfigDict(env_file=".env", env_nested_delimiter="__", env_prefix="fiab__")
 
     general: GeneralSettings = Field(default_factory=GeneralSettings)
+    product: ProductSettings = Field(default_factory=ProductSettings, description="Product specific settings")
+
     auth: AuthSettings = Field(default_factory=AuthSettings)
 
     db: DatabaseSettings = Field(default_factory=DatabaseSettings)
     api: BackendAPISettings = Field(default_factory=BackendAPISettings)
     cascade: CascadeSettings = Field(default_factory=CascadeSettings)
 
+    @classmethod
+    def settings_customise_sources(
+        cls,
+        settings_cls: type[BaseSettings],
+        init_settings: PydanticBaseSettingsSource,
+        env_settings: PydanticBaseSettingsSource,
+        dotenv_settings: PydanticBaseSettingsSource,
+        file_secret_settings: PydanticBaseSettingsSource,
+    ) -> tuple[PydanticBaseSettingsSource, ...]:
+        return env_settings, file_secret_settings, dotenv_settings, TomlConfigSettingsSource(settings_cls, fiab_home / "config.toml"), init_settings
+
+    def _get_toml(self, skip_secrets: bool = True, **k) -> str:
+        json_config = self.model_dump(mode="json", **k)
+
+        if skip_secrets:
+            (json_config.get('auth', {}).get('oidc', {}) or {}).pop('client_secret', None)
+            json_config.get('auth', {}).pop('jwt_secret', None)
```

Comment:
we should instead recursively crawl and pop all values of type SecretStr... we have some recursive crawl already here, we can generalize

Reply:
or we can have a custom child of SecretStr with overridden ser as in https://github.com/pydantic/pydantic/issues/6778, thats neat too